packages/mobile/src/pincode/authentication.ts

```diff
@@ -0,0 +1,151 @@
+import { sha256 } from 'ethereumjs-util'
+import { asyncRandomBytes } from 'react-native-secure-randombytes'
+import { PincodeType } from 'src/account/reducer'
+import { pincodeTypeSelector } from 'src/account/selectors'
+import CeloAnalytics from 'src/analytics/CeloAnalytics'
+import { CustomEventNames } from 'src/analytics/constants'
+import { navigate, navigateBack } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import {
+  getCachedPassword,
+  getCachedPasswordHash,
+  getCachedPepper,
+  getCachedPin,
+  setCachedPasswordHash,
+  setCachedPepper,
+} from 'src/pincode/PasswordCache'
+import { ensureCorrectPin } from 'src/pincode/utils'
+import { store } from 'src/redux/store'
+import { retrieveStoredItem, storeItem } from 'src/storage/keychain'
+import Logger from 'src/utils/Logger'
+import { currentAccountSelector } from 'src/web3/selectors'
+
+const TAG = 'pincode/authentication'
+enum STORAGE_KEYS {
+  PEPPER = 'PEPPER',
+  PASSWORD_HASH = 'PASSWORD_HASH',
+}
+const PEPPER_LENGTH = 64
+export const PIN_LENGTH = 6
+
+export function isPinValid(pin: string) {
+  return pin.length === PIN_LENGTH
+}
+
+function pinStorageKey(account: string) {
```

Comment:
The name of this confused me. It's actually the account password hash storage key right?